nodes/core/core/lib/debug/debug-utils.js

```diff
@@ -508,10 +506,26 @@ RED.debug = (function() {
         }
     }
 
+    function clearMessageList() {
+        $(".debug-message").remove();
+        config.clear();
+        clearFilterSettings();
```

Comment:
We do not want to clear the filter settings when the clear button is clicked in the UI - which is what this change will do.

I suggest adding a boolean argument to `clearMessageList` to indicate that the filter should also be clearer. When called from clicking the button, pass in `false`. When called because the project has changed, pass in `true`.

Reply:
Thanks @knolleary, this is actually my fault. I'll add a boolean argument to clearMessageList function.